includes/class-give-session.php

```diff
@@ -453,11 +453,13 @@ public function maybe_start_session() {
 	 * @param string  $wpUserLogin
 	 */
 	public function startSessionWhenLoginAsWPUser( $wpUserLogin, $wpUser ) {
+
 		$donor = Give()->donors->get_donor_by( 'user_id', $wpUser->ID );
 
 		// Setup session only if donor exist for specific WP user.
 		if ( $donor ) {
 			$this->maybe_start_session();
+			$this->set( 'give_email', $donor->email );
```

Comment:
@henryholtgeerts, can you please talk through the thought behind setting the email rather than always starting the session? Happy to discuss on a call if that's helpful.

Reply:
@JasonTheAdams Happy to explain -- I had the same thought initially. If you take a look at the logic to "start" a session, you find that it never actually sets the `give_email` attribute, all that starting a session entails is setting an expiry date for the cookie. Here, I opted to specifically set the `give_email` attribute since ultimately what matter is that a valid session has started (this is handled by maybe_start_session()), and that the `give_email` attribute is set to mirror the email of the currently logged in WP user. 

Future work on the Give Session class could be served well by ensuring that a simple donor ID is made available (to eliminate a need to query a Donor by their email), but without significant refactoring to other parts of the codebase it makes the most sense to simply set the `give_email` attribute.

Reply:
Excellent explanation, thanks!